### PR TITLE
Improving database logging by adding duration, referrer and post data

### PR DESCRIPTION
**Pull request by mistercrunch**
@bkyryliuk @ascott @vera-liu

### CODE REVIEW

File: superset/models.py [hunk 1]
```diff
@@ -2276,15 +2281,18 @@ def wrapper(*args, **kwargs):
...
             except:
                 pass
+            value = f(*args, **kwargs)
             log = cls(
                 action=f.__name__,
                 json=params,
                 dashboard_id=d.get('dashboard_id') or None,
                 slice_id=slice_id,
+                duration_ms=(datetime.now() - start_dttm).total_seconds() * 1000,
+                referrer=request.referrer[:1000],
```
Comment (vera-liu) [inline superset/models.py]: just curious, what do we use referrer for?

**Comment by kingo55** [comment]
Not sure if this is the right forum for suggesting this, but would it be better to allow users to log activity to Google Analytics, Snowplow Analytics or Google tag manager? 

Most webmasters are very familiar with monitoring site performance and activity in there. Would reduce the writes on the database too... 

If people need more detailed logs they could store them to disk via Nginx.

Thoughts?

File: superset/models.py [hunk 2]
```diff
@@ -2276,15 +2281,18 @@ def wrapper(*args, **kwargs):
                 params = json.dumps(d)
             except:
                 pass
+            value = f(*args, **kwargs)
             log = cls(
                 action=f.__name__,
                 json=params,
                 dashboard_id=d.get('dashboard_id') or None,
                 slice_id=slice_id,
+                duration_ms=(datetime.now() - start_dttm).total_seconds() * 1000,
```
Comment (bkyryliuk) [inline superset/models.py]: Line too long (81 > 79 characters)

**Comment by bkyryliuk** [comment]
@kingo55 sounds like they are 2 different kinds of logs. This table is used in the superset internally to provide some insights for the creators:
* views per dashboards
* views per slice etc
And it can be used to personalize user experience
Having all this data outside the service will make things a bit more complicated.

Providing a way to load the data to external services - is pretty good idea, however we do not have such a use case internally

**Comment by bkyryliuk** [comment]
LGTM once build is green

**Review by bkyryliuk — APPROVED** [review]

File: superset/models.py (hunk 1 above)
Comment (mistercrunch) [inline superset/models.py]: It has the information of the previous url in the browser that got the person there. This is neat to track people coming from other domains.

**Comment by mistercrunch** [comment]
@kingo55 I'm in sync with @bkyryliuk on this. Would happily review a PR that added a hook for Google Analytics and the likes.

I've also been meaning to ship our internal "Superset Usage" dashboard along with the codebase examples so that people can analyze how Superset is doing in their environment.

**Pull request merged**